### PR TITLE
Convert the iOS build script into github workflow steps

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -45,9 +45,82 @@ jobs:
...
+          set -euo pipefail
+
           rustup target add aarch64-apple-darwin aarch64-apple-ios aarch64-apple-ios-sim
-          sh ./scripts/ios/release.sh
+
+          echo "🚢 Switch 'useLocalFramework' to 'false' in Package.swift for release"
+          sed -i '' 's/let useLocalFramework = true/let useLocalFramework = false/' Package.swift
+
+          # output is: "<CHKSUM>;<$XCFRAME_ZIP_PATH>"
+          OUTPUT_OF_BUILD=$(bash scripts/ios/build-sargon.sh --release-tag "$NEXT_TAG" | tail -n 1) || exit $?
```
Comment (CyonAlexRDX) [inline .github/workflows/release.yml:70]: I guess you wanna migrate over the contents of `build-sargon.sh` too?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -14,40 +14,103 @@ env:
...
+          git status
+          git commit -m "Release of '$NEXT_TAG'" \
+                     -m "Updated Package.swift with new checksum and path to zip on Github, and maybe apple/Sources/UniFFI/Sargon.swift." \
+                     -m "This commit is not merged into main branch (and need not be)." \
+                     --no-verify
+
+          echo "🚢 🏷️ 📡 Pushing tag: $NEXT_TAG, but only tag, not commit."
+          git push origin $NEXT_TAG
         env:
+          NEXT_TAG: ${{ steps.tags.outputs.NEXT_TAG }}
```
Comment (CyonAlexRDX) [inline .github/workflows/release.yml:101]: same here, do we need to add `needs` perhaps ?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -14,40 +14,103 @@ env:
...
+
+      - name: Push release tag
+        if: github.ref == 'refs/heads/main'
+        run: |
+          set -euo pipefail
+
+          # We have .gitigored Sargon.swift because we dont need it in git history, but we
+          # need it for this release, so we must FORCE add it (since it is ignored).
+          echo "🚢  Staging changed files"
+          git add --force Package.swift apple/Sources/UniFFI/Sargon.swift
```
Comment (CyonAlexRDX) [inline .github/workflows/release.yml:86]: @muzuke are you certain that the git state between jobs are shared? I thought jobs ran in parallell? Here we run `git add` which will only work if this job (lacks ID btw...?) `Push release tag` "shares git/file state" with the job `build`.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -14,40 +14,103 @@ env:
...
+      - name: Create release
+        if: github.ref == 'refs/heads/main'
         run: |
-          rustup target add aarch64-apple-darwin aarch64-apple-ios aarch64-apple-ios-sim
-          sh ./scripts/ios/release.sh
+          # This MUST match whatever you we have declared in Package.swift
+          SWIFT_SARGON_BINARY_ASSET_NAME="libsargon-rs.xcframework.zip"
+
+          GH_RELEASE_TITLE="v$NEXT_TAG"
+          gh release create $NEXT_TAG '$XCFRAME_ZIP_PATH#$SWIFT_SARGON_BINARY_ASSET_NAME' --generate-notes --notes-start-tag $LAST_TAG --title '$GH_RELEASE_TITLE'
```
Comment (CyonAlexRDX) [inline .github/workflows/release.yml:111]: same as https://github.com/radixdlt/sargon/pull/267/files#r1846424837 this job will only work if the build artifact from job `build` is "shared" with this step.

**Review by CyonAlexRDX — APPROVED** [review]
LGTM

**Review by alpeto9 — APPROVED** [review]

**Pull request merged**